src/modules/person/persistence/person.repository.ts

```diff
@@ -405,21 +401,16 @@ export class PersonRepository {
         return mapEntityToAggregate(personEntity);
     }
 
-    private hasUsernameChanged(
-        oldVorname: string,
-        oldFamilienname: string,
-        newVorname: string,
-        newFamilienname: string,
-    ): boolean {
-        const oldVornameLowerCase: string = oldVorname.toLowerCase();
-        const oldFamiliennameLowerCase: string = oldFamilienname.toLowerCase();
-        const newVornameLowerCase: string = newVorname.toLowerCase();
-        const newFamiliennameLowerCase: string = newFamilienname.toLowerCase();
+    private hasChangedNames(personEntity: PersonEntity, person: Person<true>): boolean {
+        const oldVorname: string = personEntity.vorname.toLowerCase();
+        const oldFamilienname: string = personEntity.familienname.toLowerCase();
+        const newVorname: string = person.vorname.toLowerCase();
+        const newFamilienname: string = person.familienname.toLowerCase();
 
         //only look for first letter, because username is firstname[0] + lastname
-        if (oldVornameLowerCase[0] !== newVornameLowerCase[0]) return true;
+        if (oldVorname[0] !== newVorname[0]) return true;
 
-        return oldFamiliennameLowerCase !== newFamiliennameLowerCase;
+        return oldFamilienname !== newFamilienname;
     }
```

Comment:
I thought we said the renamed-event should be fired if the name changed at all.

That would mean this (to me at least):
- Don'd lowercase the names (changes in casing should count as an updated name)
- compare the full first and last name

Reply:
This should be discussed with Corneluis because i see that the EmailEventHandler only needs to know if the person needs a new username after name update.

Reply:
@phaelcg @marode-cap Can I merge this?

Reply:
@YoussefBouch i will merge it

Reply:
Das Problem mit hasChangedNames() für PersonRenamedEvent  wird mit SPSH-1049 gelöst.